src/components/Molecules/Task/index.jsx

```diff
@@ -0,0 +1,65 @@
+import React,{useState} from "react";
+import {Checkbox} from "../../Atoms/Checkbox"
+import Input from "../../Atoms/Input"
+import {EditButton} from "../../Atoms/EditButton"
+import styled from 'styled-components';
+import COLOR from "../../../variables/color";
+import TEXT from "../../../variables/texts";
+import FONTFAMILY from "../../../variables/font_family";
+
+export default function Task({onTaskNameChange,onTaskComplete,taskName = "",defaultIsEditing=false,}) {
+
+  const [isEditing, setIsEditing] = useState(defaultIsEditing);
+
+  const onEditButtonClick = () => {
+    setIsEditing(true);
+  };
+
+  const onEditComplete = (value) => {
+    setIsEditing(false);
+    onTaskNameChange(value);
+  }
+
+
+  return (
+    <StyledWrapper>
+      <StyledCheckboxWrapper>
+        <Checkbox onClick={onTaskComplete}/>
+      </StyledCheckboxWrapper>
+      <StyledNameAndButtonWrapper>
+        {isEditing ? (<Input onEditComplete={onEditComplete} defaultValue={taskName} />) : (<StyledTaskName>{taskName}</StyledTaskName>)}
+        <StyledEditButtonWrapper>{isEditing ? ("") : (<EditButton onClick={onEditButtonClick}/>)}</StyledEditButtonWrapper>
```

Comment:
どうせisEditingで評価するなら、こっちの方が良いんじゃない？

```suggestion
        {isEditing ? (<Input onEditComplete={onEditComplete} defaultValue={taskName} />) : (<StyledTaskName>{taskName}</StyledTaskName>
        <StyledEditButtonWrapper><EditButton onClick={onEditButtonClick}/></StyledEditButtonWrapper>)}
```